### PR TITLE
Simplify pan map

### PR DESCRIPTION
**Pull request by peterbourgon**
Further usage would be like

```bash
$ pan map 10.0.*.* | jq 'select(.os == "freebsd")' | tee freebsd.txt
{"ip": "10.0.1.1", "hostname": "lb.int.net", "os": "freebsd", "arch": "x86_64"}
$ cat freebsd.txt | pan top  # or whatever
```

### CODE REVIEW

**Comment by peterbourgon** [comment]
This would be implemented by embedding `facts` binaries (one per supported OS/arch) into `pan`, which would be remotely executed on all discovered, ssh-able hosts. `facts` would just output the JSON object to stdout. It would optionally take as an argument the IP address used to access the host, i.e. `facts 10.0.1.1`.

File: README.md [hunk 1]
```diff
@@ -5,33 +5,31 @@ Pan is *distributed* orchestration with **digestable** output.
...
-usage: pan map <hosts> [flags]
+usage:
+  pan map <hosts> [flags]
 
 flags:
-  -bastion="127.0.0.1": Bastion host
-  -batch=false: Batch results
-  -hops=3: Max number of hops
   -key="~/.ssh/id_rsa": SSH key to use
-  -port=22: SSH port
```
Comment (tsenart) [inline README.md]: I suppose we can add this later. I was over designing at this point. :-)

File: README.md [hunk 2]
```diff
@@ -5,33 +5,31 @@ Pan is *distributed* orchestration with **digestable** output.
...
+
 ```shell
 $ pan map -h
-usage: pan map <hosts> [flags]
+usage:
+  pan map <hosts> [flags]
 
 flags:
-  -bastion="127.0.0.1": Bastion host
-  -batch=false: Batch results
```
Comment (tsenart) [inline README.md]: What are the semantics of data output of this command? Do we wait for all the hosts to be contacted and then output everything or do we stream things out as they come?

File: README.md [hunk 3]
```diff
@@ -5,33 +5,31 @@ Pan is *distributed* orchestration with **digestable** output.
...
-This works by scanning the provided `<hosts>` from the `-bastion` host
-for the given `-port`, connecting to it over SSH, gathering facts and
-repeating the process on each reachable host that runs a supported OS
-and architecture. The `-hops` flag defines the maximum number of times this process happens.
-The `<hosts>` syntax is simple. It supports ranges and wild cards.  i.e. `10.0.[0-128].*`
+
+`map` collects facts about machines in your network.
+This works by scanning the provided `<hosts>` expression.
+The `<hosts>` expression syntax is simple.
+It supports ranges and wild cards, i.e. `10.0.[0-28].*`
```
Comment (tsenart) [inline README.md]: Nice and concise. You have your way with words :)

Comment (tsenart) [inline README.md]: The details of **how** things happen should be documented well somewhere else at some point (a man page?)

Comment (tsenart) [inline README.md]: This is answering the right question in the right place: **what**?

File: README.md (hunk 2 above)
Comment (peterbourgon) [inline README.md]: Streamed as they arrive, stopping when we've exhausted the hosts represented by the expression.

Comment (tsenart) [inline README.md]: Cool. That's what `batch=false` meant.

Comment (tsenart) [inline README.md]: But yeah, we don't need two different semantics for now. Although, I `-timeout`s will be indispensable.

Comment (peterbourgon) [inline README.md]: Yeah, `-timeout` should be easy to compose in.

**Pull request merged**